wasm/jit/engine.go

```diff
@@ -434,40 +400,39 @@ func (e *engine) exec(f *compiledWasmFunction) {
 			// Meaning that the current frame exits
 			// so restore the caller's frame.
 			e.callFramePop()
-		case jitCallStatusCodeCallWasmFunction:
-			// This never panics as we made sure that the index exists for all the referenced functions
-			// in a module.
-			nextFunc := e.compiledWasmFunctions[e.functionCallIndex]
-			// Calculate the continuation address so we can resume this caller function frame.
-			currentFrame.continuationAddress = currentFrame.wasmFunction.codeInitialAddress + e.continuationAddressOffset
-			currentFrame.continuationStackPointer = e.stackPointer + nextFunc.resultCount - nextFunc.paramCount
-			// Create the callee frame.
-			frame := &callFrame{
-				continuationAddress: nextFunc.codeInitialAddress,
-				wasmFunction:        nextFunc,
-				// Set the base pointer to the beginning of the function params
-				stackBasePointer: e.stackBasePointer + e.stackPointer - nextFunc.paramCount,
+		case jitCallStatusCodeCallFunction:
+			nextFunc := e.compiledFunctions[e.functionCallAddress]
+			if nextFunc.isHostFunction() {
+				currentFrame.continuationAddress = currentFrame.compiledFunction.codeInitialAddress + e.continuationAddressOffset
```

Comment:
since this is the same in both branches, move it up? that let's us see the differences easier.